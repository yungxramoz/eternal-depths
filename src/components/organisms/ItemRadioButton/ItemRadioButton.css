.item-radio-label {
  height: auto !important;
}
