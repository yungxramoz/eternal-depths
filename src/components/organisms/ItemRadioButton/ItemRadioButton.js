import React from 'react'
import ItemDetails from '../../molecules/ItemDetails/ItemDetails'
import './ItemRadioButton.css'

const ItemRadioButton = ({
  onChange,
  disabled,
  icon,
  title,
  subtitle,
  description,
  id,
  checked,
}) => {
  return (
    <>
      <input
        disabled={disabled}
        className="rpgui-radio golden"
        name="radio-golden"
        id={id}
        value={id}
        checked={checked}
        type="radio"
        onChange={onChange}
      />
      <label className="item-radio-label" htmlFor={id}>
        <ItemDetails
          title={title}
          subtitle={subtitle}
          description={description}
          icon={icon}
        />
      </label>
    </>
  )
}

export default ItemRadioButton
