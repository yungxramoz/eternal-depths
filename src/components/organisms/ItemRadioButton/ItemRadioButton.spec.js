import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import ItemRadioButton from './ItemRadioButton'

describe('ItemRadioButton', () => {
  it('renders correctly', () => {
    render(
      <ItemRadioButton
        onChange={() => {}}
        disabled={false}
        icon="test-icon"
        title="test-title"
        subtitle="test-subtitle"
        description="test-description"
        id="test-id"
        checked={false}
      />,
    )
    expect(document.body).toMatchSnapshot('ItemRadioButton')
  })

  it('calls onChange prop when clicked', () => {
    const onChange = jest.fn()
    render(
      <ItemRadioButton
        onChange={onChange}
        disabled={false}
        icon="test-icon"
        title="test-title"
        subtitle="test-subtitle"
        description="test-description"
        id="test-id"
        checked={false}
      />,
    )
    fireEvent.click(screen.getByRole('radio'))
    expect(onChange).toHaveBeenCalled()
  })
})
