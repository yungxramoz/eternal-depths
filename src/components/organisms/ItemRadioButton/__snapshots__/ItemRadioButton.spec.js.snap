// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRadioButton renders correctly: ItemRadioButton 1`] = `
<body>
  <div>
    <input
      class="rpgui-radio golden"
      id="test-id"
      name="radio-golden"
      type="radio"
      value="test-id"
    />
    <label
      class="item-radio-label"
      for="test-id"
    >
      <div
        class="item-container"
      >
        <div
          class="rpgui-icon test-icon item-icon"
        />
        <div
          class="item-content"
        >
          <p
            class="item-title"
          >
            test-title
          </p>
          <p
            class="item-subtitle"
          >
            <i>
              test-subtitle
            </i>
          </p>
          <p
            class="item-description"
          >
            test-description
          </p>
        </div>
      </div>
    </label>
  </div>
</body>
`;
